Could not find Rust_CARGO_CACHED using the following names: cargo

Just trying this out on a project in NixOS.
My cmake:
```
add_subdirectory(mod/corrosion)
find_package(Corrosion REQUIRED)
```
Output:
```
$ cmake ..
CMake Error at mod/corrosion/cmake/FindRust.cmake:186 (find_program):
  Could not find Rust_CARGO_CACHED using the following names: cargo
Call Stack (most recent call first):
  mod/corrosion/cmake/Corrosion.cmake:22 (find_package)
  mod/corrosion/CMakeLists.txt:63 (include)
```
I see that cargo executable is looked up in `HINTS "${_RUST_TOOLCHAIN_PATH}/bin"`. On my system, cargo isn't placed anywhere near Rust:
```
$ which rustc
/nix/store/sqw563f9ggjjkzlrf37h6fvqiz40hwpj-rustc-1.57.0/bin/rustc

$ which cargo
/nix/store/5q06j36iqzymwsgskq3p7lidl8bfkjxv-cargo-1.57.0/bin/cargo
```

--- a/cmake/FindRust.cmake
+++ b/cmake/FindRust.cmake
@@ -181,13 +181,25 @@ if (NOT Rust_COMPILER_CACHED)
     return()
 endif()
 
-# Look for Cargo next to rustc.
-# If you want to use a different cargo, explicitly set `Rust_CARGO` variable
-find_program(
-    Rust_CARGO_CACHED
-    cargo
-        HINTS "${_RUST_TOOLCHAIN_PATH}/bin"
-        REQUIRED NO_DEFAULT_PATH)
+if (_RESOLVE_RUSTUP_TOOLCHAINS)
+    # If not already explicitly set by the `Rust_CARGO` variable, search for cargo next to rustc,
+    # since the toolchain is managed by rustup.
+    # Note: Not all toolchains managed by rustup have `cargo` installed. Specifically, a custom
+    # toolchain may not have had cargo built. In this case the user should manually specify
+    # `Rust_CARGO`.
+    find_program(
+        Rust_CARGO_CACHED
+        cargo
+            HINTS "${_RUST_TOOLCHAIN_PATH}/bin"
+            REQUIRED NO_DEFAULT_PATH)
+else()
+    # On some systems (e.g. NixOS) cargo is not managed by rustup and also not next to rustc.
+    find_program(
+            Rust_CARGO_CACHED
+            cargo
+                HINTS "${_RUST_TOOLCHAIN_PATH}/bin"
+                REQUIRED)
+endif()
 
 set(CARGO_RUST_FLAGS "" CACHE STRING "Flags to pass to rustc")
 set(CARGO_RUST_FLAGS_DEBUG "" CACHE STRING

